include/mbedtls/ccm.h

```diff
@@ -110,7 +109,7 @@ int mbedtls_ccm_setkey( mbedtls_ccm_context *ctx,
  * \brief   This function releases and clears the specified CCM context
  *          and underlying cipher sub-context.
  *
- * \param ctx       The CCM context to clear.
+ * \param ctx       The CCM context to clear. Must not be NULL.
```

Comment:
Well actually, `xxx_free()` functions should accept `NULL` and just do nothing - most of them already do, and in case some users rely on that, we don't want to break their expectations (even if it was never documented before).

Reply:
In this case, the `ctx` variable is dereferenced inside the function. It will crash for a `NULL` pointer. So we can either validate input in such cases, or, for consistency, add an `if` that will silently return upon receiving `NULL` argument. I have reservations about the second approach, but it may be the best in the context of the rest of the library.

What do you think we should do here?

Reply:
What we usually do for `xxx_free()` functions is to start them with:
```
if( ctx == NULL )
    return;
```
It's a debatable approach, but it's what we do most of the time in the library so we should do it here as well for consistency.

Reply:
I agree.